lib/Runtime/Types/DynamicObject.cpp

```diff
@@ -295,6 +295,24 @@ namespace Js
         this->type = type;
     }
 
+    void DynamicObject::ReplaceTypeToPredecessorType(DynamicType * predecessorType)
+    {
+        Assert(this->GetTypeHandler()->IsPathTypeHandler());
+        Assert(((PathTypeHandlerBase*)this->GetTypeHandler())->GetPredecessorType()->GetTypeHandler()->IsPathTypeHandler());
+
+        Assert(((PathTypeHandlerBase*)this->GetTypeHandler())->GetPredecessorType() == predecessorType);
+
+        Assert(!predecessorType->isLocked || predecessorType->GetTypeHandler()->GetIsLocked());
+        Assert(!predecessorType->isShared || predecessorType->GetTypeHandler()->GetIsShared());
+
+        PathTypeHandlerBase* currentPathTypeHandler = (PathTypeHandlerBase*)this->GetTypeHandler();
+        PathTypeHandlerBase* predecessorPathTypeHandler = (PathTypeHandlerBase*)predecessorType;
+
+        Assert(predecessorPathTypeHandler->GetInlineSlotCapacity() >= currentPathTypeHandler->GetInlineSlotCapacity());
```

Comment:
>decessorPathTypeHandler->GetInlineSlotCapacity() >= currentPathTypeHandler [](start = 18, length = 74)

Can we never have case where currentPathType has more inlineSlotCapacity than predecessor except the shrink slot case?